src/webgl/p5.Framebuffer.js

```diff
@@ -931,6 +950,238 @@ class Framebuffer {
     callback();
     this.end();
   }
+
+  loadPixels() {
+    const gl = this.gl;
+    const prevFramebuffer = gl.getParameter(gl.FRAMEBUFFER_BINDING);
+    gl.bindFramebuffer(gl.FRAMEBUFFER, this.framebuffer);
+    const colorFormat = this._glColorFormat();
+    this.pixels = readPixelsWebGL(
+      this.pixels,
+      gl,
+      this.framebuffer,
+      0,
+      0,
+      this.width * this.density,
+      this.height * this.density,
+      colorFormat.format,
+      colorFormat.type
+    );
+    gl.bindFramebuffer(gl.FRAMEBUFFER, prevFramebuffer);
+  }
+
+  /**
+   * Get a region of pixels, or a single pixel, from the canvas.
+   *
+   * Returns an array of [R,G,B,A] values for any pixel or grabs a section of
+   * an image. If the Framebuffer has been set up to not store alpha values, then
+   * only [R,G,B] will be returned. If no parameters are specified, the entire
+   * image is returned.
+   * Use the x and y parameters to get the value of one pixel. Get a section of
+   * the display window by specifying additional w and h parameters. When
+   * getting an image, the x and y parameters define the coordinates for the
+   * upper-left corner of the image, regardless of the current <a href="#/p5/imageMode">imageMode()</a>.
+   *
+   * @method get
+   * @param  {Number}         x x-coordinate of the pixel
+   * @param  {Number}         y y-coordinate of the pixel
+   * @param  {Number}         w width
+   * @param  {Number}         h height
+   * @return {p5.Image}       the rectangle <a href="#/p5.Image">p5.Image</a>
+   */
+  /**
+   * @method get
+   * @return {p5.Image}      the whole <a href="#/p5.Image">p5.Image</a>
+   */
+  /**
+   * @method get
+   * @param  {Number}        x
+   * @param  {Number}        y
+   * @return {Number[]}      color of pixel at x,y in array format [R, G, B, A]
+   */
+  get(x, y, w, h) {
+    const colorFormat = this._glColorFormat();
+    if (x === undefined && y === undefined) {
+      x = 0;
+      y = 0;
+      w = this.width;
+      h = this.height;
+    } else if (w === undefined && h === undefined) {
+      if (x < 0 || y < 0 || w >= this.width || h >= this.height) {
+        return [0, 0, 0, 0];
```

Comment:
Would a warning help the programmer in this case? An alternate behavior to consider would be to clamp the edge pixel. Another alternate would be to modulo the pixel location with the width or height. What do you think would be the most expected?

Reply:
I think maybe clamping + a warning would be good.

Also, note to myself when implementing these changes, `w` and `h` should be `x` and `y` here